test-network-function/generic/suite.go

```diff
@@ -346,6 +358,116 @@ func testGracePeriod(context *interactive.Context, podName, podNamespace string)
+	currentKernelCmdlineArgsTester := currentkernelcmdlineargs.NewCurrentKernelCmdlineArgs(defaultTimeout)
+	test, err := tnf.NewTest(targetPodOc.GetExpecter(), currentKernelCmdlineArgsTester, []reel.Handler{currentKernelCmdlineArgsTester}, targetPodOc.GetErrorChannel())
+	gomega.Expect(err).To(gomega.BeNil())
+	testResult, err := test.Run()
+	gomega.Expect(testResult).To(gomega.Equal(tnf.SUCCESS))
+	gomega.Expect(err).To(gomega.BeNil())
+	currnetKernelCmdlineArgs := currentKernelCmdlineArgsTester.GetKernelArguments()
+	currentSplitKernelCmdlineArgs := strings.Split(currnetKernelCmdlineArgs, " ")
+	return argListToMap(currentSplitKernelCmdlineArgs)
+}
+
+func getGrubKernelArgs(context *interactive.Context, nodeName string) map[string]string {
+	bootConfigTester := bootconfig.NewBootConfig(defaultTimeout, nodeName)
+	test, err := tnf.NewTest(context.GetExpecter(), bootConfigTester, []reel.Handler{bootConfigTester}, context.GetErrorChannel())
+	gomega.Expect(err).To(gomega.BeNil())
+	testResult, err := test.Run()
+	gomega.Expect(testResult).To(gomega.Equal(tnf.SUCCESS))
+	gomega.Expect(err).To(gomega.BeNil())
+	bootConfig := bootConfigTester.GetBootConfig()
+	splitBootConfig := strings.Split(bootConfig, "\n")
+	filteredBootConfig := filterArray(splitBootConfig, func(line string) bool {
+		return strings.HasPrefix(line, "options")
+	})
+	gomega.Expect(len(filteredBootConfig)).To(gomega.Equal(1))
+	grubKernelConfig := filteredBootConfig[0]
+	grubSplitKernelConfig := strings.Split(grubKernelConfig, " ")
+	grubSplitKernelConfig = grubSplitKernelConfig[1:]
+	return argListToMap(grubSplitKernelConfig)
+}
+
+func testBootParams(context *interactive.Context, podName, podNamespace string, targetPodOc *interactive.Oc) {
+	ginkgo.It(fmt.Sprintf("Testing boot params for the pod's node %s/%s", podNamespace, podName), func() {
+		nodeName := getPodNodeName(context, podName, podNamespace)
+		mcName := getMcName(context, nodeName)
+		mcKernelArgumentsMap := getMcKernelArguments(context, mcName)
+		currentKernelArgsMap := getCurrentKernelCmdlineArgs(targetPodOc)
+		grubKernelConfigMap := getGrubKernelArgs(context, nodeName)
+
+		mcKernelArgumentsKeys := reflect.ValueOf(mcKernelArgumentsMap).MapKeys()
+		for _, arg := range mcKernelArgumentsKeys {
+			argKey := arg.String()
+			if val, ok := currentKernelArgsMap[argKey]; ok {
+				gomega.Expect(val).To(gomega.Equal(mcKernelArgumentsMap[argKey]))
+			}
+			if val, ok := grubKernelConfigMap[argKey]; ok {
+				gomega.Expect(val).To(gomega.Equal(grubKernelConfigMap[argKey]))
+			}
+		}
+	})
+}
+
+func argListToMap(lst []string) map[string]string {
+	retval := make(map[string]string)
+	for _, arg := range lst {
+		splitArgs := strings.Split(arg, "=")
+		if len(splitArgs) == 1 {
+			retval[splitArgs[0]] = ""
+		} else {
+			retval[splitArgs[0]] = splitArgs[1]
+		}
+	}
+	return retval
+}
+
+func filterArray(vs []string, f func(string) bool) []string {
```

Comment:
`filterArray` and `argListToMap` are at a very different level of abstraction to the rest of the file and as such feel out of place where they are. Please move them to another file, and add a small comment explaining their purpose.

Reply:
Moved to "pkg/utils/utils.go"